accounts/keystore/keystore.go

```diff
@@ -312,11 +312,9 @@ func (ks *KeyStore) Unlock(a accounts.Account, passphrase string) error {
 // Lock removes the private key with the given address from memory.
 func (ks *KeyStore) Lock(addr common.Address) error {
 	ks.mu.Lock()
+	defer ks.mu.Unlock()
 	if unl, found := ks.unlocked[addr]; found {
-		ks.mu.Unlock()
```

Comment:
This changes the semantics. Previously, it released the lock before invoking `ks.expire`, but with your changes it retains the lock during the call. 

Ostensibly, this was coded like it is for a reason. Please motivate why you consider this change safe/needed

Reply:
If you just want to remove some code, but retain the semantics, you could do 
```golang
    ks.mu.Lock() 
    unl, found := ks.unlocked[addr]
    ks.mu.Unlock()
    if found{
        ks.expire(addr, unl, time.Duration(0)*time.Nanosecond)
    }
    return nil
```